scripts/rmc.py

```diff
@@ -73,49 +89,67 @@ def cargo_build(crate, verbose=False, debug=False):
-
+    return run_cmd(cmd, label="to-gotoc", verbose=verbose)
 
 def run_cbmc(cbmc_filename, cbmc_args, verbose=False, quiet=False):
     cbmc_cmd = ["cbmc"] + cbmc_args + [cbmc_filename]
-    process = subprocess.Popen(cbmc_cmd, universal_newlines=True,
-                               stdout=subprocess.PIPE,
-                               stderr=subprocess.STDOUT)
-    if not quiet:
-        for line in iter(process.stdout.readline, ""):
-            print(line, end="")
-    process.stdout.close()
-    retcode = process.wait()
-    print_rmc_step_status("cbmc", process, verbose)
+    return run_cmd(cbmc_cmd, label="cbmc", output_to="stdout", verbose=verbose, quiet=quiet)
+
+def run_visualize(cbmc_filename, cbmc_args, verbose=False, quiet=False, keep_temps=False, function="main", srcdir=".", wkdir="."):
+    results_filename = "results.xml"
+    coverage_filename = "coverage.xml"
+    property_filename = "property.xml"
+    temp_goto_filename = "temp.goto"
+    if not keep_temps:
+        for filename in [results_filename, coverage_filename, property_filename, temp_goto_filename]:
+            atexit.register(delete_file, filename)
+
+    # 1) goto-cc --function main <cbmc_filename> -o temp.goto
+    # 2) cbmc --xml-ui --trace ~/rmc/library/rmc/rmc_lib.c temp.goto > results.xml
+    # 3) cbmc --xml-ui --cover location ~/rmc/library/rmc/rmc_lib.c temp.goto > coverage.xml
+    # 4) cbmc --xml-ui --show-properties ~/rmc/library/rmc/rmc_lib.c temp.goto > property.xml
+    # 5) cbmc-viewer --result results.xml --coverage coverage.xml --property property.xml --srcdir . --goto temp.goto
+
+    run_goto_cc(cbmc_filename, temp_goto_filename, verbose, quiet)
+    
+    def run_cbmc_local(cbmc_args, output_to):
+        cbmc_cmd = ["cbmc"] + cbmc_args + [temp_goto_filename]
+        return run_cmd(cbmc_cmd, label="cbmc", output_to=output_to, verbose=verbose, quiet=quiet)
+
+    cbmc_xml_args = cbmc_args + ["--xml-ui"]
+    retcode = run_cbmc_local(cbmc_xml_args + ["--trace"], results_filename)
+    run_cbmc_local(cbmc_xml_args + ["--cover", "location"], coverage_filename)
+    run_cbmc_local(cbmc_xml_args + ["--show-properties"], property_filename)
+
+    run_cbmc_viewer(temp_goto_filename, results_filename, coverage_filename, property_filename, verbose, quiet, srcdir, wkdir)
+
     return retcode
 
+def run_goto_cc(src, dst, verbose=False, quiet=False, function="main"):
+    cmd = ["goto-cc"] + ["--function", function] + [src] + ["-o", dst]
+    return run_cmd(cmd, label="goto-cc", verbose=verbose, quiet=quiet)
+
+def run_cbmc_viewer(goto_filename, results_filename, coverage_filename, property_filename, verbose=False, quiet=False, srcdir=".", wkdir="."):
+    cmd = ["cbmc-viewer"] + \
+          ["--result", results_filename] + \
+          ["--coverage", coverage_filename] + \
+          ["--property", property_filename] + \
+          ["--srcdir", os.path.abspath(srcdir)] + \
+          ["--wkdir", os.path.abspath(wkdir)] + \
```

Comment:
I'm not sure if this is how you get "physical path" (I couldn't find anything about it in Python docs or online related to Python).

Reply:
I think it is realpath